diff -r -u binutils-2.13.1.orig/bfd/elf32-v850.c binutils-2.13.1/bfd/elf32-v850.c
--- binutils-2.13.1.orig/bfd/elf32-v850.c	Fri Jan 17 21:16:21 2003
+++ binutils-2.13.1/bfd/elf32-v850.c	Sun Jan 19 21:24:58 2003
@@ -1532,15 +1532,6 @@
   symtab_hdr = & elf_tdata (input_bfd)->symtab_hdr;
   sym_hashes = elf_sym_hashes (input_bfd);
 
-  if (sym_hashes == NULL)
-    {
-      info->callbacks->warning
-	(info, "no hash table available",
-	 NULL, input_bfd, input_section, (bfd_vma) 0);
-
-      return false;
-    }
-
   /* Reset the list of remembered HI16S relocs to empty.  */
   free_hi16s     = previous_hi16s;
   previous_hi16s = NULL;
